File: session_py/src/session_py/globals_test.py
import session_py.globals as globals


def test_globals_initial_values():
    assert globals.SCALE == 1e6
    assert globals.PI == 3.141592653589793
    assert globals.ANGLE == 0.11
    assert globals.TOLERANCE == 1e-3


def test_globals_modification():
    original_scale = globals.SCALE
    globals.SCALE = 2000.0
    assert globals.SCALE == 2000.0
    globals.SCALE = original_scale

File: session_cpp/src/globals_test.cpp
#include "catch/include/catch_amalgamated.hpp"
#include "globals.h"

using namespace geo;

TEST_CASE("test_globals_initial_values") {
    REQUIRE(GLOBALS::SCALE == 1e6);
    REQUIRE(GLOBALS::PI == 3.14159265358979323846);
    REQUIRE(GLOBALS::ANGLE == 0.11);
    REQUIRE(GLOBALS::TOLERANCE == 1e-3);
}

TEST_CASE("test_globals_modification") {
    double original_scale = GLOBALS::SCALE;
    GLOBALS::SCALE = 2000.0;
    REQUIRE(GLOBALS::SCALE == 2000.0);
    GLOBALS::SCALE = original_scale;
}
